File: openflash_user/openflash_front/src/lib/appLog.test.js
import test from 'node:test'
import assert from 'node:assert/strict'
import { mock } from 'node:test'
import { appError, appWarn } from './appLog.js'

test('appError calls console.error with code prefix', () => {
  const calls = []
  mock.method(console, 'error', (...args) => calls.push(args))
  appError(40010, '卡片操作失败')
  assert.equal(calls.length, 1)
  assert.ok(calls[0][0].includes('[E:40010]'), `expected [E:40010] in: ${calls[0][0]}`)
  mock.restoreAll()
})

test('appError includes original message', () => {
  const calls = []
  mock.method(console, 'error', (...args) => calls.push(args))
  appError(50000, '服务器错误')
  assert.ok(calls[0][0].includes('服务器错误'))
  mock.restoreAll()
})

test('appWarn calls console.warn with code prefix', () => {
  const calls = []
  mock.method(console, 'warn', (...args) => calls.push(args))
  appWarn(60001, '不支持 SSE')
  assert.equal(calls.length, 1)
  assert.ok(calls[0][0].includes('[E:60001]'))
  mock.restoreAll()
})

test('appError forwards extra args', () => {
  const calls = []
  mock.method(console, 'error', (...args) => calls.push(args))
  const err = new Error('cause')
  appError(50000, '失败', err)
  assert.equal(calls[0][1], err)
  mock.restoreAll()
})

File: openflash_user/openflash_back/src/test/java/openflash_core/common/AppLogTest.java
package openflash_core.common;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.ArgumentCaptor;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;
import org.slf4j.Logger;

import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.mockito.ArgumentMatchers.any;
import static org.mockito.ArgumentMatchers.anyString;
import static org.mockito.Mockito.verify;

@ExtendWith(MockitoExtension.class)
class AppLogTest {

    @Mock
    private Logger log;

    @Test
    void errorIncludesCodeNameInFormat() {
        AppLog.error(log, ErrorCode.GENERIC_ERROR, "something went wrong");
        ArgumentCaptor<String> fmtCaptor = ArgumentCaptor.forClass(String.class);
        verify(log).error(fmtCaptor.capture(), any(Object[].class));
        assertTrue(fmtCaptor.getValue().contains("GENERIC_ERROR"),
            "format should contain code name");
    }

    @Test
    void errorIncludesCodeValueInFormat() {
        AppLog.error(log, ErrorCode.GENERIC_ERROR, "something went wrong");
        ArgumentCaptor<String> fmtCaptor = ArgumentCaptor.forClass(String.class);
        verify(log).error(fmtCaptor.capture(), any(Object[].class));
        assertTrue(fmtCaptor.getValue().contains("50000"),
            "format should contain code value 50000");
    }

    @Test
    void warnIncludesCodeNameInFormat() {
        AppLog.warn(log, ErrorCode.UPLOAD_FILE_DELETE_FAILED, "file delete failed");
        ArgumentCaptor<String> fmtCaptor = ArgumentCaptor.forClass(String.class);
        verify(log).warn(fmtCaptor.capture(), any(Object[].class));
        assertTrue(fmtCaptor.getValue().contains("UPLOAD_FILE_DELETE_FAILED"),
            "format should contain code name");
    }

    @Test
    void errorPreservesOriginalMessageInFormat() {
        AppLog.error(log, ErrorCode.CARD_NOT_FOUND, "card {} not found");
        ArgumentCaptor<String> fmtCaptor = ArgumentCaptor.forClass(String.class);
        verify(log).error(fmtCaptor.capture(), any(Object[].class));
        assertTrue(fmtCaptor.getValue().contains("card {} not found"),
            "format should contain original message");
    }

    @Test
    void errorPassesThroughExtraArgs() {
        RuntimeException ex = new RuntimeException("cause");
        AppLog.error(log, ErrorCode.GENERIC_ERROR, "failed", ex);
        ArgumentCaptor<Object[]> argsCaptor = ArgumentCaptor.forClass(Object[].class);
        verify(log).error(anyString(), argsCaptor.capture());
        Object[] args = argsCaptor.getValue();
        boolean hasException = false;
        for (Object a : args) {
            if (ex.equals(a)) { hasException = true; break; }
        }
        assertTrue(hasException, "extra args should be forwarded");
    }
}
